src/js/view/view.js

```diff
@@ -340,9 +350,59 @@ function View(_api, _model) {
         playerViewModel.change('playlistItem', onPlaylistItem);
         // Triggering 'resize' resulting in player 'ready'
         _lastWidth = _lastHeight = null;
+
+        // Setup floating scroll handler
+        if (_floatingConfig && OS.mobile) {
+            _model.change('inDom', (model, val) => {
+                if (val) {
+                    viewsManager.addScrollHandler(throttledMobileFloatScrollHandler);
+                } else {
+                    viewsManager.removeScrollHandler(throttledMobileFloatScrollHandler);
+                }
+            });
+        }
+
         this.checkResized();
     };
 
+    // Functions for handler float on scroll (mobile)
+    const FLOATING_TOP_OFFSET = 62;
+    let canFire = true;
+    let debounceTO;
+    function checkFloatOnScroll() {
+        const floating = _model.get('isFloating');
+        const enoughRoomForFloat = playerBounds.top < FLOATING_TOP_OFFSET;
+        const hasCrossedThreshold = enoughRoomForFloat ?
+            playerBounds.top <= window.scrollY :
+            playerBounds.top <= window.scrollY + FLOATING_TOP_OFFSET;
+
+        if (!floating && hasCrossedThreshold) {
+            _updateFloating(0, enoughRoomForFloat);
+        } else if (floating && !hasCrossedThreshold) {
+            _updateFloating(1, enoughRoomForFloat);
+        }
+    }
+
+    function throttledMobileFloatScrollHandler() {
+        if (!fosMobileBehavior()) {
```

Comment:
just like how the below `updateFloating` is responsible for checking whether or not we should follow this floating behavior, i used that same pattern here to deal with the mobile portrait vs. landscape issue (landscape acts like desktop but up to this point it's all been handled in css)